packages/react-tokens/src/variablesByFile.js

```diff
@@ -0,0 +1,269 @@
+              bestMatch = key;
+              bestValue = localVarsMap[match][key];
+            }
+          }
+        }
+      }
+      if (!bestMatch) {
+        // eslint-disable-next-line no-console
+        console.error(`no matching selector found for ${match} in localVarsMap`);
+      }
+      return bestValue;
+    }
+  } else {
+    // eslint-disable-next-line no-console
+    console.error(`no matching property found for ${match} in localVarsMap`);
+  }
+};
+
+const getComputedCssVarValue = (value, selector) =>
+  value.replace(/var\(([\w|-]*)\)/g, (full, match) => {
+    if (match.startsWith('--pf-global')) {
+      if (cssGlobalsToScssVarsMap[match]) {
+        return cssGlobalsToScssVarsMap[match];
+      } else {
+        return full;
+      }
+    } else {
+      if (selector) {
+        return getFromLocalMap(match, selector);
+      }
+    }
+  });
+
+const getFinalValue = (value, selector) =>
+  value.replace(/var\(([\w|-]*)\)/g, (full, match) => {
+    if (match.startsWith('--pf-global')) {
+      if (cssGlobalVariablesMap[match]) {
+        return cssGlobalVariablesMap[match];
+      } else {
+        return full;
+      }
+    } else {
+      if (selector) {
+        return getFromLocalMap(match, selector);
+      }
+    }
+  });
+
+const getComputedScssVarValue = value =>
+  value.replace(/\$pf[^,)\s*/]*/g, match => {
+    if (combinedScssVarsColorsMap[match]) {
+      return combinedScssVarsColorsMap[match];
+    } else {
+      return match;
+    }
+  });
+
+const getVarsMap = (value, selector) => {
+  // evaluate the value and follow the variable chain
+  let varsMap = [value];
+
+  let computedValue = value;
+  let finalValue = value;
+  while (
+    finalValue.indexOf('var(--pf') > -1 ||
+    computedValue.indexOf('var(--pf') > -1 ||
+    computedValue.indexOf('$pf-') > -1
+  ) {
+    // keep following the variable chain until we get to a value
+    if (finalValue.indexOf('var(--pf') > -1) {
+      finalValue = getFinalValue(finalValue, selector);
+    }
+    if (computedValue.indexOf('var(--pf') > -1) {
+      computedValue = getComputedCssVarValue(computedValue, selector);
+    } else {
+      computedValue = getComputedScssVarValue(computedValue);
+    }
+    varsMap.push(computedValue);
+  }
+  const lastElement = varsMap[varsMap.length - 1];
+  if (lastElement.indexOf('pf-') > -1) {
+    varsMap.push(finalValue);
+  }
+  // all values should not be boxed by var()
+  varsMap = varsMap.map(variable => variable.replace(/var\(([\w|-]*)\)/g, (full, match) => match));
+
+  return varsMap;
+};
+
+const tokens = {};
+cssFiles.forEach(filePath => {
+  const absFilePath = resolve(pfStylesDir, filePath);
+  const cssAst = parse(readFileSync(absFilePath, 'utf8'));
+  // key is the formatted file name, e.g. c_about_modal_box
+  const key = formatFilePathToName(filePath);
+
+  cssAst.stylesheet.rules.forEach(node => {
+    if (node.type !== 'rule' || node.selectors.indexOf('.pf-t-dark') !== -1) {
```

Comment:
`.filter().forEach()` is a little more readable.

Reply:
agree, changed it